### PR TITLE
Ran/assign lesson modal

### PR DESCRIPTION
**Pull request by stormcrow85**

### CODE REVIEW

**Comment by assafshp** [comment]
jenkins-test

**Comment by orenor10** [comment]
jenkins-test

**Comment by orenor10** [comment]
Jenkins-test

**Comment by assafshp** [comment]
jenkins-test

**Comment by assafshp** [comment]
jenkins-test

File: Gruntfile.js [hunk 1]
```diff
@@ -89,9 +108,12 @@ module.exports = function (grunt) {
             }
         },
         concat: {
-            dist: {
-                src: ['<%= yeoman.src %>/core/*.js', '<%= yeoman.src %>/components/**/module.js', '<%= yeoman.src %>/components/**/*.js','.tmp/*.js'],
-                dest: '<%= yeoman.dist %>/<%= yeoman.appName %>.js'
+            mainModule: {
```
Comment (assafshp) [inline Gruntfile.js]: where is dist ????

File: Gruntfile.js [hunk 2]
```diff
@@ -236,9 +313,82 @@ module.exports = function (grunt) {
         }
         console.log('serving from', grunt.config('connect').options.base);
         grunt.task.run([
-            'sass',
+            'build',
```
Comment (assafshp) [inline Gruntfile.js]: where is sass?

File: src/components/assign-lesson/assign-lesson.controller.js [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+        .controller('assignLessonCtrl', function ($scope, $q, locals, SubjectEnum, $translate, $translatePartialLoader, $log, $http) {
+            'ngInject';
+
+            //$translatePartialLoader.addPart('assign-lesson');
+            //$translate.refresh();
+
+            $scope.vm = {};
+            $scope.vm.cssClass = locals.cssClass;
+            $scope.vm.modalTitle = 'Assign Lesson';
+            $scope.vm.currentStudent = 'Brandon Butler'; // TODO: mock
```
Comment (assafshp) [inline src/components/assign-lesson/assign-lesson.controller.js]: MOCK DATA cannot be in infra

File: src/components/assign-lesson/assign-lesson.controller.js [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+            //            "date":1466674319530,
+            //            "guid":"79db996b-cd75-404b-b913-1f883520fea2",
+            //            "moduleId":2,
+            //            "uid":"a04aebb4-50e9-48f8-a8db-4964c4350b84",
+            //            "tutorId":null
+            //        }
+            //    }
+            //];
+
+            $http({
```
Comment (assafshp) [inline src/components/assign-lesson/assign-lesson.controller.js]: mock cannot be in infra

File: src/components/assign-lesson/assign-lesson.controller.js [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+            $scope.vm.currentSubject = '';
+            $scope.vm.currentStatus = '';
+
+
+            /**
+             * Filters
+             */
+
+            $scope.vm.subjects = [
+                {
```
Comment (assafshp) [inline src/components/assign-lesson/assign-lesson.controller.js]: how does it fit for SAT and ACT?

File: src/components/assign-lesson/assign-lesson.controller.js [hunk 6]
```diff
@@ -0,0 +1,261 @@
...
+                    name: 'Not Assigned',
+                    val: false
+                }
+            ];
+
+            //$translate([
+            //    translateNamespace + '.' + 'EXERCISE_TYPES.EXERCISES',
+            //    translateNamespace + '.' + 'EXERCISE_TYPES.SECTIONS'
+            //]).then(function (translation) {
+            //    self.exerciseTypesForFilter = [
```
Comment (assafshp) [inline src/components/assign-lesson/assign-lesson.controller.js]: ? NEEDED?

File: src/components/assign-lesson/assign-lesson.controller.js [hunk 7]
```diff
@@ -0,0 +1,261 @@
...
+                        if (obj.id === selectedStatusId) {
+                            self.currentSelectedStatus = obj.name;
+                        }
+                    });
+                }
+            };
+
+
+            $scope.vm.submitAssigned = function() {
+              console.log('assigned button clicked');
```
Comment (assafshp) [inline src/components/assign-lesson/assign-lesson.controller.js]: no console log -> $log.

File: src/components/assign-lesson/lessonList.directive.js [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+                                var filter = filters[i];
+                                if (!filter(item)) {
+                                    return;
+                                }
+                            }
+                            scope.d.processedData.push(item);
+                        });
+                        //
+                        var t1 = performance.now();
+                        console.log("Call to filterData took " + (t1 - t0) + " milliseconds.");
```
Comment (assafshp) [inline src/components/assign-lesson/lessonList.directive.js]: not in code.
in any way I prefer console.time and console.timeend

**Pull request closed without merging**